pkg/network/ebpf/c/protocols/kafka/kafka-classification.h

```diff
@@ -268,6 +268,44 @@ static __always_inline bool validate_first_topic_name(pktbuf_t pkt, bool flexibl
     CHECK_STRING_VALID_TOPIC_NAME(TOPIC_NAME_MAX_STRING_SIZE_TO_VALIDATE, topic_name_size, topic_name);
 }
 
+// Reads the first topic id (can be multiple) from the given offset,
+// verifies if it is a valid UUID version 4
+static __always_inline bool validate_first_topic_id(pktbuf_t pkt, bool flexible, u32 offset) {
+    // The topic id is a UUID, which is 16 bytes long.
+    // It is in network byte order (big-endian)
+    u8 topic_id[16] = {};
+
+    // Skipping number of entries for now
+    if (flexible) {
+        if (!skip_varint_number_of_topics(pkt, &offset)) {
+            return false;
+        }
+    } else {
+        offset += sizeof(s32);
+    }
+
+    if (offset + sizeof(topic_id) > pktbuf_data_end(pkt)) {
+        return false;
+    }
+
+    pktbuf_load_bytes(pkt, offset, topic_id, sizeof(topic_id));
+    offset += sizeof(topic_id);
+
+    // The UUID version (13th digit)
+    if (topic_id[6] >> 4 != 0x4) {
```

Comment:
perhaps it will be clearer to do `topic_id[6] && 0x40 == 0x40`

Reply:
same for the other condition

Reply:
I actually rolled it back, I like the `>> 4` syntax better as I can just do `__u8 uuid_version = topic_id[6] >> 4;` and use it like a normal number so `uuid_version == 4`
It feels safer because you don't have to remember anything for checks later